WikiWatershed/model-my-watershed
# Add BMP sidebar and connect to ScenarioModel

This PR adds a sidebar/popup that shows the BMPs that have been added and connects the BMP UI to the ScenarioModel, so that adding BMPs triggers running the GWLF-E model with the modifications. It also makes some other related changes to the UI.

![screen shot 2016-05-27 at 1 06 15 pm 2](https://cloud.githubusercontent.com/assets/1896461/15616370/4c8c0942-2411-11e6-96f2-ddcf885d35de.png)
![screen shot 2016-05-27 at 1 06 18 pm 2](https://cloud.githubusercontent.com/assets/1896461/15616371/4c9079e6-2411-11e6-9310-35f34166ba4a.png)
![screen shot 2016-05-27 at 1 06 23 pm 2](https://cloud.githubusercontent.com/assets/1896461/15616372/4c916644-2411-11e6-8395-a1c7bd497b2a.png)

##### Testing
* Run `./scripts/debugcelery.sh`
* For a GWLF-E model, try adding BMPs. The sidebar should appear when there are modifications. Check that the last two BMPs work.
* Look in the console for the data that is sent for the GWLF-E run after the BMP is added. (I found that Chrome crashed when trying to look at the payload in the Network tab). Verify that the correct variables are set. For instance, 'Infiltration' should set `PctAreaInfil` to the fraction (0-1) computed from the input and `Qretention` to 2.54.
* Check that the popup show correct data, delete button works (and triggers GWLF-E run), and that it closes when clicking outside or on close button.
* Check that 'Conservation Practices' doesn't appear until Mapshed results are back, and that 'Compare' button is hidden.
* When logged in, check that BMPs are saved and show up after hitting refresh in the model view.
* Check that TR55 and Compare view still work.

Connects #1316 
Connects #1279 
Connects #1280
